website/docs/flink_configuration.md

```diff
@@ -3,115 +3,177 @@ title: Flink Setup
+
+You can configured the execution mode via the command line:
+
+```sh
+$ bin/flink run -Dexecution.runtime-mode=BATCH <jarFile>
+
+```
+
+Separately, you can programmatically create and configure the `StreamExecutionEnvironment`, a Flink programming API. This execution environment is how all data pipelines are created and maintained.
+
+You can configure the execution mode programmatically. Below is an example of how to set the `BATCH` mode.
+
+```sh
+StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+env.setRuntimeMode(RuntimeExecutionMode.BATCH);
+```
+See the [Flink docs](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/datastream/execution_mode/) for more details.
+
+## Global Configurations​
+
+The global configurations are used to tune Flink for throughput, memory management and/or checkpoints (disaster recovery i.e., data loss). Two of the most important global configurations for a Flink job are parallelism and memory. For a long-running job, the initial resource configuration is crucial because open-source Flink does not support auto-pilot yet, where you can automatically scale up or down resources when there’s high or low data ingestion. So, you might waste or underutilize resources. 
+
+All Hudi-specific parallelism and memory configurations depend on your Flink job resources.
+
+When using Flink, you can set some global configurations in `$FLINK_HOME/conf/flink-conf.yaml`.
+
+### Parallelism​
+
+If your system has a lot of data to ingest, increasing the parallelism can improve throughput significantly. Hudi supplies flexible config options for specific operators, but at a high level, a default global parallelism can reduce the complexity of manual configuration. Try the default configuration and adjust as necessary. 
+
+| Property Name | Default  | Description | Scope | Since Version                          |
+|----------------|--------|----------|---------------|--------------------------------------|
+| `taskmanager.numberOfTaskSlots` | 1 | The is the number of parallel operators or user function instances that a single TaskManager can run. We recommend setting this value > 4, and the actual value needs to be set according to the amount of data | n/a | 0.9.0 |
+| `parallelism.default` | 1 | The is the default parallelism used when no parallelism is specified anywhere (default: 1). For example, if the value of [`write.bucket_assign.tasks`](#parallelism-1) is not set, this value will be used | n/a | 0.9.0.|
+
+### Memory​
+The `JobManager` and `TaskManager` memory configuration is very important for a Flink job to work smoothly. Below, we'll describe these configurations. 
+
+#### JobManager
+The JobManager handles all the instants coordination. It keeps an in-memory fs view for all the file handles on the filesystem within its embedded timeline server. We need to ensure enough memory is allocated to avoid OOM errors. The configs below allow you to allocate the necessary memory. 
```

Comment:
```suggestion
The JobManager handles all the instants coordination. It keeps an in-memory fs view for all the file handles on storage within an embedded timeline server. We need to ensure enough memory is allocated to avoid OOM errors. The configs below allow you to allocate the necessary memory. 
```

Reply:
"instants coordination" is a bit unclear